packages/cli/src/build.ts

```diff
@@ -198,7 +206,28 @@ function buildSingleFile(
         log('[Build]', 'output transpiled css');
         tryRun(() => fs.writeFileSync(cssAssetOutPath, cssCode), `Write File Error: ${outPath}`);
     }
-    // .d.ts?
+    // .d.ts
+    if (dts) {
+        const dtsContent = createModuleSource(res, 'dts', false);
+
+        log('[Build]', 'output .d.ts');
+        tryRun(
+            () => fs.writeFileSync(outSrcPath + '.d.ts', dtsContent),
+            `Write File Error: ${outPath}`
+        );
+
+        // .d.ts.map
+        // if not explicitly defined, assumed true with "--dts" parent scope
+        if (dtsSourceMap === undefined || dtsSourceMap === true) {
+            const dtsMappingContent = createModuleSource(res, 'dts.map', false);
```

Comment:
if `createModuleSource` would accept options or would provide separate functions for each module type, then I would prefer passing the created `d.ts` rather then recreating it internally

Reply:
will open issue

Reply:
opened #1843